src/core/public/map_to_object.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function mapToObject(map: Map<string, unknown>) {
+  const result: Record<string, unknown> = Object.create(null);
+  for (const [key, value] of map) {
```

Comment:
`Object.fromEntries(map.entries())`?

Reply:
I tried. but TS told me this is too fresh syntax 😄 

another interesting case for you guys, TS allows me to use `[].flatMap`, because:
```
  // Node 8 should support everything output by esnext, we override this
  // in webpack with loader-level compiler options
  "target": "esnext",
```
but it's not implemented in node v10, they added it in v11.
so our TS setup is a bit misleading